### ISSUE TITLE
 When fullscreen, switching active windows, feels glitchy

### ISSUE THREAD

**Issue by joelmathewthomas** [hyprwm/Hyprland#8564]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland 0.45.0 built from branch  at commit 8fa4cfb7dfa0be52a66db5f3b13e0ecde3d78b41  (keybinds: don't animate fullscreen size/pos changes coming in).
Date: Mon Nov 11 19:25:37 2024
Tag: v0.45.0-9-g8fa4cfb7, commits: 5442
built against aquamarine 0.5.0


flags set:
debug


System Information:
System name: Linux
Node name: linux
Release: 6.11.9-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Sun, 17 Nov 2024 16:06:17 +0000


GPU information: 
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation AD107M [GeForce RTX 4050 Max-Q / Mobile] [10de:28e1] (rev a1) (prog-if 00 [VGA controller])
35:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Rembrandt [Radeon 680M] [1002:1681] (rev 0a) (prog-if 00 [VGA controller])
NVRM version: NVIDIA UNIX Open Kernel Module for x86_64  565.57.01  Release Build  (archlinux-builder@)  


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:


```

</details>

### Description

After commit 8fa4cfb7dfa0be52a66db5f3b13e0ecde3d78b41, when fullscreen and switching activewindows, normal windows look fine , but web apps like whatasapp, etc.. look weird, because all that's seen is the web interface trying to fix it's viewport size by glitching. Earlier , when the whole window had an animation as if it was transitioning to fullscreen, the gltiching of the webpage didn't seem all that bad. Now there is no animation while switching the window. All that's there is the web interface glitching to change it's viewport size.

### How to reproduce

install commit 8fa4cfb7dfa0be52a66db5f3b13e0ecde3d78b41.
and try switching interfaces like whatsapp web

**Comment by vaxerski** [comment]
using ff looks normal?

**Comment by joelmathewthomas** [comment]
> using ff looks normal?

No, even with firefox, with websites like YouTube and WhatsApp web, when switching windows all that's seen, is the window changing its viewport quickly and the website glitching with no smooth transitions, to be responsive.

**Comment by joelmathewthomas** [comment]
https://github.com/user-attachments/assets/345a237c-e81d-4d35-8a99-e12cb7e666c8

**Comment by adamnejm** [comment]
@vaxerski This is because when a new fullscreen aplication is focused, the current one is tiled back into the layout. You can see this very clearly by extending the animation duration:
```
animations = {
  bezier = ease-in-out, 0.6, 0.0, 0.4, 1.0
  animation = fadeIn,  1, 20.00, ease-in-out
  animation = fadeOut, 1, 20.00, ease-in-out
}
```
Which looks like this:

https://github.com/user-attachments/assets/1546ff5a-c9d6-4be1-9f54-d7fa2d19f229

This is especially annoying, because disabling these specific animations, or even global animations is not enough for the graphical gitch to disappear. There's 1 frame where the client resizing is visible and it hurts. This video is with animations disabled, at 25% playback speed:

https://github.com/user-attachments/assets/5cf7b72a-8b25-4dd6-8983-a36b389b79bf

Of course the best solution would be to not never actually resize / tile back the fullscreen clients, but display them on top of each other...

PS. Would be really nice to have a separate animation for switching between fs'd clients.

**Comment by adamnejm** [comment]
This also seems to utterly break fullscreen behavior if the (game) window cannot be resized, and is responsible for "cursor snapping", to the floor in games (tested Minecraft and Garry's Mod).

**Comment by vaxerski** [comment]
if that's what this is about, then #683

**Comment by tomkhoailang** [comment]
try hyprland-plugins/hyprscrolling with column_width = 1.0. problem solved, don't know it'll work as exptected with game or not
